src/Queen.cpp

```diff
@@ -0,0 +1,20 @@
+#include "Queen.h"
+
+Queen::Queen(char name, int row, int column, string player, Board* board) : Rook(name, row, column, player, board), Bishop(name, row, column, player, board), Piece(name, row, column, player, board)
+{
+
+}
+
+Queen::~Queen()
+{
+}
+
+int Queen::isValidMove(Location start, Location end)
+{
+	int code_rook = Rook::isValidMove(start,end);
```

Comment:
Intermediate classes must be created for walking diagonally and walking straight, thus giving a finer inheritance for different tools. then you do have to do this